tools/ml-dataprep/queries.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module containing the SQL query templates used to extract ML datasets from
+BigQuery."""
+
+# Query template to count the total number of lines in a table.
+QUERY_COUNT_TEMPLATE = """
+	SELECT COUNT(*) AS lines FROM `{0}`
+"""
+
+# Query template to shuffle the source data and save it in a temporary table.
+QUERY_TEMP_DATA_TEMPLATE = """
+	CREATE OR REPLACE TABLE
+	  `{0}`
+	AS
+	SELECT
+		CAST(RAND() * {4} AS INT64) AS random,
+		*
+	FROM (
+		SELECT
+			{1}, {2}
+		FROM
+			`{3}`
+	)
+"""
```

Comment:
when you have many parameters you may want to name them. It will help readability and catching mistakes

```python
QUERY_TEMP_DATA_TEMPLATE = "CREATE TABLE {dest_table} AS SELECT * FROM {target_table}"

query_str = QUERY_TEMP_DATA_TEMPLATE(dest_table='foo', target_table='bar')
```

this way you don't need to remember order of parameters

Reply:
Done in commit 46c2f5a64057049903dbbf5c6caa93863f8b2ca2.